app/controllers/courses_controller.rb

```diff
@@ -26,5 +26,9 @@ def show
     @subscription = (current_user.subscriptions.find_by(course_slug: @course.slug) if current_user?)
 
     @vote_total = Vote.get_course_vote_total(@course.slug)
+
+    @reactions = current_user.note_reactions.where(course_slug: @course.slug).map { |r| [r.lesson_slug, r.reaction_name] }.to_h if current_user?
+
+    @progress = current_user.progresses.where(course_slug: @course.slug).map(&:lesson_slug) if current_user?
```

Comment:
There are now (at least 3 different data objects that get initialised if there's a `current_user` present – may be worth sticking these all in a `if current_user? … end` block to contain them in one place?